### ISSUE TITLE
More control over Page and System Layout

### ISSUE THREAD

**Issue by rpatters1** [musescore/MuseScore#25107]
### Your idea

This is a two-pronged request to allow for final polish and control over Page and System layout.

## Add a per-system, per-part **System properties** feature. ##

At a minimum it would include:

- a Spatium percentage relative to the Score-level style setting. (Alternatively, though to my mind less desirable, a separate per-system Spatium setting.)
- a list of all staves in the score or part with the ability to hide or show any empty staves on a per system basis.

Nice-to-haves would be

- a way to specify extra space for one or more particular staves on the system. (Even better would be a way to drag them WYSIWYG.)
- potentially other settings from the Staff properties that would apply only on this system in this score/part.

I'm not sure whether a Properties panel or a separate dialog or some other UI makes the most sense. The properties panel is already crowded, and this would be a fairly tall panel, since it has to list all the instruments. What seems essential is that each change would need to be reflected in real-time.

## The ability to force measures per system and systems per page. ##

Beyond this, I would also like the ability to force measures per system and systems per page. MuseScore could highlight collisions in a highly color. (Even better it could allow navigation to each collision so that the user could manually address them.) The request is related to #23792, and I am of the opinion that the "Add" part of an Add/Remove page breaks feature would be much more useful if the number of systems per page could be forced.

### Problem to be solved

MuseScore's automatic page layout is very impressive, especially to someone like me coming from Finale. However, like any automated system before the future epoch when our AI overlords take over, it only gets us most of the way there. MuseScore needs to allow for final polish by the user. Some of the polishing requirements that appear to me to be difficult to achieve are

1. The top staff line and the bottom staff line on every page should be in the same vertical location. (Top and bottom margins for the music that align with the top and bottom staff lines respectively on every page.)
2. Fitting large orchestral tuttis onto a single page within the margins as described in 1.
3. Filling out systems with empty staves to achieve better balance on a particular page. (And specifically, some systems should be full score, regardless of empty staves.)
4. If no other layout requirements prevent it, facing pages of a conductor score should have the same staff layout. This is a human judgement call and is basically covered by 3. I suppose this is just a plea to allow for copying one system layout to the next.

Once a polished system layout is in place, there should be a way to lock it into place. At that point the polished layout is more important than collisions caused by future edits. Once the layout is locked, I would like to be notified of collisions rather than see my polished layout destroyed by the robot.

### Prior art

Since Finale has no automated layout, the user has to do everything. But it is both possible to achieve a polished layout and to lock it into place. And with the help of plugins, the polished layout is not such a terrible task.

As for prior art in the literature, late 19th and 20th century score abound in examples of pages showing different system percentages to accommodate larger orchestral tuttis.

1. Nearly all engraved editions that I've looked at show the top staff line and bottom staff line at the same vertical location on every page. (Unless there is text to push the top down, such as movement titles.)
2. Many late 19th and 20th century editions reduce the system size to fit orchestral tuttis on a page according to 1.  _Le Sacre du Printemps_ always provides good examples, but so do Mahler and Strauss scores.
3. I leafed through the Ravel "Four Orchestral Works" Dover edition and found a number of examples where they pad the system with empty systems until it looks like it is the full ensemble but in fact omits a percussion instrument or a 2nd harp or somethng. I also found this example in _Le Sacre_ shown below. The 1st violins empty staff is added to the second system to allow its staff spacing to balance with the top system.

<img width="635" alt="Screen Shot 2024-10-09 at 9 03 34 AM" src="https://github.com/user-attachments/assets/530f8d11-e46c-4bce-8d2f-994cf7323916">

4. One of my plugins copied system layouts from one page to the next, so this was really easy in Finale. I suppose at worse case this could be a plugin in MuseScore as well, if a future plugin API provides the hooks to do it.

### Additional context

I am still a novice MuseScore user, but I have done a lot of due diligence to ascertain if any of these requests are already possible. If I missed something, please forgive me. But if I *did* miss something, it was fairly well hidden.

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by rpatters1** [comment]
A further thought. Perhaps the documented issues could be solved with the first two bullets above. The lack of per-system staff hiding and per-system spatium percentage or setting a showstoppers for me, at least. The rest is nice-to-have. Maybe the locking isn't even that nice. I can be persuaded either way.

**Comment by tygoee** [comment]
> The ability to force measures per system and systems per page.

This is already possible since Musescore 4.1, see the [documentation](https://musescore.org/en/handbook/4/systems-and-horizontal-spacing#keep-measures-on-the-same-system)

**Comment by rpatters1** [comment]
I don't see forcing systems-per-page at that doc website, but I'm also not sure it's that important. I'm thinking about closing this issue and resubmitting it with what's important:

- the ability to show empty staves on a per-system basis
- the ability to set a spatium size for a system different (usually smaller) than the score spatium size.

**Comment by tygoee** [comment]
You're right, that isn't quite an option but also achievable using layout spacers, format>page options>staff space and margins.

The issue you mentioned (the one I filed) was the one where I had found this issue from in similar issues, and then replied. It's not really the same though. I was requesting an unhide staff option, and you are requesting a complete layout interface overhaul, where I don't specifically see something about unhiding hidden staves with 'Hide empty measures'.

**Comment by tygoee** [comment]
About the plugins: I've never made a plugin myself, but after seeing [this plugin](https://musescore.org/en/project/export-layout-breaks-parts), I think you're able to do that. The code is short and you can view it when you open the `.qml` file. I don't know if you have any experience making plugins though.

**Comment by rpatters1** [comment]
The second bullet above is explicitly what you are asking for.

**Comment by mike-spa** [comment]
Closing this issue as most of the points have been either already addressed or are covered by other more specific issues.
